packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1983,53 +1991,89 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         proposalHandle: string | undefined,
         ackHandle: string,
         summaryLogger: ITelemetryLogger,
-        version?: IVersion,
     ) {
-        const getSnapshot = async () => {
-            const perfEvent = PerformanceEvent.start(summaryLogger, {
+        const readAndParseBlob = async <T>(id: string) => readAndParse<T>(this.storage, id);
+        await this.summarizerNode.refreshLatestSummary(
+            proposalHandle,
+            async () => this.fetchSnapshotFromStorage(ackHandle, summaryLogger, {
```

Comment:
Not specific to this PR, but it feels like code would be more readable if
1. refreshLatestSummary is broken into two APIs - one taking handle, and one taking snapshot & sequence number
2. Later API would be providing actual content (snapshots, seq) instead of callbacks how to get them.